### ISSUE TITLE
can't assign tuple(list[T]) to tuple[T, ...] in generic types

### ISSUE THREAD

**Issue by hmk114** [microsoft/pyright#9937]
**Describe the bug**
I got a strange error when checking the following code snippet:

```python
from typing import TypeVar, Generic

T = TypeVar("T")


class A(Generic[T]):
    def f(self):
        lst, value = self.g()

        a: tuple[T, ...] = tuple(lst)
        dct: dict[tuple[T, ...], float] = {}
        b: tuple[T, ...] = ()
        dct[a] = 1.0
        dct[b] = 1.0

    def g(self):
        lst: list[T] = []
        value = 234
        x = "anything"

        for _ in range(1):
            if x:
                value = x
                break

        return lst, value
```

output of command `pyright test.py`:
```text
test.py
  test.py:13:9 - error: Argument of type "tuple[T@A, ...]" cannot be assigned to parameter "key" of type "tuple[T@A, ...]" in function "__setitem__"
    "builtins.tuple" is not assignable to "builtins.tuple"
      Tuple entry 1 is incorrect type
        Type "T@A" is not assignable to type "T@A" (reportArgumentType)
1 error, 0 warnings, 0 informations
```

I can't understand why `dct[b] = 1.0` is ok here but `dct[a] = 1.0` is not. Maybe I'm wrong and this is not a bug. Could someone please explain it to me? Many thanks!

**VS Code extension or command-line**
Command line tool (installed via `pip install pyright`):
- pyright version: 1.1.394
- python version: 3.12.0

I've tested vscode extension (version 2025.2.100), and the error above is also present.

**Comment by erictraut** [comment]
Thanks for the bug report. This is indeed a bug in pyright's return type inference logic. It will be addressed in the next release.

In the meantime, you can work around it by adding an explicit return type annotation to method `g` in your example.

**Comment by hmk114** [comment]
Thanks for your response! Happy to know the bugfix😊
